enzyme/Enzyme/Enzyme.cpp

```diff
@@ -133,8 +133,140 @@ handleCustomDerivative(llvm::Module &M, llvm::GlobalVariable &g,
           }
         }
 
+        std::set<size_t> byref;
+
         if (Mode == DerivativeMode::ReverseModeGradient) {
-          assert(numargs == 3);
+          assert(numargs >= 3);
+          for (size_t i = numargs; i < CA->getNumOperands(); i++) {
```

Comment:
It would be nice if this could be encapsulated in its own function if possible, so that we can reuse this in BatchMode.